include/glow/Runtime/RuntimeTypes.h

```diff
@@ -0,0 +1,84 @@
+/**
+ * Copyright (c) 2017-present, Facebook, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#ifndef GLOW_RUNTIME_RUNTIMETYPES_H
+#define GLOW_RUNTIME_RUNTIMETYPES_H
+
+#include "glow/Graph/Graph.h"
+
+#include <string>
+#include <unordered_map>
+#include <vector>
+
+namespace glow {
+namespace runtime {
+
+using NetworkID = size_t;
+using DeviceID = size_t;
+
+/// Enum to communicate results when communicating with device at initialization
+/// and runtime.
+enum ResultCode { READY, EXECUTED, FAILED, CANCELLED };
+
+struct DeviceMemoryInfo {
+  /// Available memory on device in MB
+  double availableMemory;
```

Comment:
I'd use `size_t` for memory, since it's not really a floating-point quantity.  I also have a slight preference for bytes (rather than a bigger unit like MB), but I also know it's kind of annoying to look at quantities like 536,870,912 instead of "512", so your discretion there.

Reply:
Actually, `size_t` might be a little awkward since it's based on the host's address space.  In practice I don't think it would ever matter, but maybe a `uint64_t` would be better?

Reply:
changed to uint64_t.